### DISCUSSION TITLE
Convert table to record

### DISCUSSION THREAD

**Discussion by fellnerse** [nushell/nushell#5032] [Q&A] [answered]
Hello! I love the version 0.60 and am in the process of migrating to it from 0.44.

I had a script to load environment variables like this:

```
let fnm-env = (fnm env --shell bash | lines |  str find-replace 'export ' '' | str find-replace -a '"' '' | split column = | rename name value | where name != "FNM_ARCH")
$fnm-env | where name != PATH | load-env
```
```
$fnm-env | where name != PATH                                                                                                                                          03/30/2022 03:16:49 PM
╭───┬───────────────────────────┬────────────────────────────────────────────────────────────────────────────╮
│ # │           name            │                                   value                                    │
├───┼───────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
│ 0 │ FNM_MULTISHELL_PATH       │ /Users/sebastianfellner/Library/Caches/fnm_multishells/84905_1648646209407 │
│ 1 │ FNM_VERSION_FILE_STRATEGY │ local                                                                      │
│ 2 │ FNM_DIR                   │ /Users/sebastianfellner/.fnm                                               │
│ 3 │ FNM_LOGLEVEL              │ info                                                                       │
│ 4 │ FNM_NODE_DIST_MIRROR      │ https://nodejs.org/dist                                                    │
╰───┴───────────────────────────┴────────────────────────────────────────────────────────────────────────────╯
```
But now I get this error, as load-env now would like to have a record. Do you know how to convert a table to record? I'm missing something like prepend for record to generate it by iterating over the table.
```
$fnm-env | where name != PATH | load-env                                                                                                                               03/30/2022 03:16:57 PM
Error: nu::shell::unsupported_input (link)

  × Unsupported input
   ╭─[entry #79:1:1]
 1 │ $fnm-env | where name != PATH | load-env
   ·                                 ────┬───
   ·                                     ╰── Record not supported
   ╰────
```

Thanks in advance!

**Comment by kubouch** [comment]
The error is confusing, can you file a bug report for that?

The `load-env` command and accepts a record, not a table now. So instead of `[[name value]; [foo bar]]` you need to pass it `{ foo: bar }`. You need to change the output to look like `{ FNM_DIR: /Users/sebastianfellner/.fnm, FNM_LOGLEVEL: info, ...}`.

**Reply by fellnerse** [reply]
I can do that! (here: https://github.com/nushell/nushell/issues/5033)
Yeah I noticed that, but I dont know how to generate a record from a dynamic list. I mean I can hard code the names, but I would prefer to do this dynamically. Do you have an idea for that?

**Comment by kubouch** [comment, marked as answer]
You can construct env variable names dynamically. You need to `reduce` the table (table is a list of records in 0.60) into a single record. Something like this: `... | reduce -f {} {|it, acc| $acc | upsert $it.column1 $it.column2 }`.

**Reply by fellnerse** [reply]
awesome, thank you very much!

**Reply by fellnerse** [reply]
Might make sense to add a reference in the docs to records

**Reply by kubouch** [reply]
We have this: https://www.nushell.sh/book/types_of_data.html#records but if you feel it's missing somewhere else, feel free to send a PR to https://github.com/nushell/nushell.github.io

We might also want bring `reduce` forward a bit more, it's essential to some programming patterns.

**Reply by fellnerse** [reply]
I meant the cross reference to reduce at exactly that point of documentation to show how records can be build programatically. I can do that

**Reply by fellnerse** [reply]
here it is: https://github.com/nushell/nushell.github.io/pull/357

**Comment by mrwensveen** [comment]
A minor drawback of `reduce` is that you need to know the column names. In this example you explicitly named them but you can convert any two-column table into a record with `$mytable | each { values } | into record`.

If your table has more than two columns you can select the first two columns with `$mytable | select ...($in | columns | get 0 1) | each { values } | into record` (you can select different columns by tweaking the `get` expression).

Examples:
```nushell
[[name value]; [foo bar] [foz baz] [aname avalue]] | each { values } | into record
[[name value comment]; [foo bar 'nice!'] [foz baz 'nize!'] [aname avalue 'nais!']] | select ...($in | columns | get 0 1) | each { values } | into record
```

both result in
```
╭───────┬────────╮
│ foo   │ bar    │
│ foz   │ baz    │
│ aname │ avalue │
╰───────┴────────╯
```
